### DISCUSSION TITLE
IntelliJ (PyCharm) Copilot plugin only suggests one line at a time

### DISCUSSION THREAD

**Discussion by martin-marko** [community/community#7318] [Copilot Conversations] [answered]
It works if I keep pressing tab and enter, but I'd rather have the full multiline suggestion as is, that'd make a lot more sense. Example:

![image](https://user-images.githubusercontent.com/82279281/140419674-8d0eac5f-a1bc-4a31-82ca-ecd7747e9a97.png)

Meanwhile in VSC:

![image](https://user-images.githubusercontent.com/82279281/140419846-1f834108-31df-472f-994d-8bba723da470.png)

**Comment by jansorg** [comment, marked as answer]
Thanks for reporting! The JetBrains plugin is also capable to suggest multiple lines. But this depends on the position in the code, and what you observed is a context where it doesn't work as well as VSCode. We're tracking this internally now to fix.

**Reply by payne911** [reply]
As a plugin developer, I would be interested in integrating our own internal ML model for code-completion.
Are you aware of which JetBrains APIs was used in the CoPilot plugin for suggesting things in gray over multiple lines after the cursor?

**Reply by RemyMachado** [reply]
This bug still occurs as december 2022. Any update?

**Comment by kimadeline** [comment]
@RemyMachado Could you provide a few more details on how this behaviour happens on your end? Sample project or code, other tabs you might have open in the editor, if it happens just for PyCharm (python code)?

Thanks!

**Reply by RemyMachado** [reply]
It's happening on IntelliJ Webstorm (TypeScript) on a big project. I get one line suggestions most of the time. If I open the GH-Copilot dedicated window it tends to suggest multi-lines more often.

**Reply by RemyMachado** [reply]
https://user-images.githubusercontent.com/30229752/208865099-3e20aea4-0114-4e25-98fc-4878e41b0b72.mp4

**Reply by kimadeline** [reply]
Thank you for the details! There was a similar discussion filed a few weeks ago on that same topic, I'd recommend you keep an eye on it as well while we investigate: https://github.com/orgs/community/discussions/40522

Thank you 🙂

**Comment by 52617365** [comment]
Still happens. Crazy lol

**Comment by MAGDALENAPLROMANOWSKA** [comment]
> It works if I keep pressing tab and enter, but I'd rather have the full multiline suggestion as is, that'd make a lot more sense. Example:
> 
> ![image](https://user-images.githubusercontent.com/82279281/140419674-8d0eac5f-a1bc-4a31-82ca-ecd7747e9a97.png)
> 
> Meanwhile in VSC:
> 
> ![image](https://user-images.githubusercontent.com/82279281/140419846-1f834108-31df-472f-994d-8bba723da470.png)

**Comment by MAGDALENAPLROMANOWSKA** [comment]
@ #

**Comment by Paul-de-Barbeyrac** [comment]
I cancelled my Github Copilot suggestion due to this poor integration with PyCharm.
Single line suggestion is a waste of time as opposed to block suggestions where you have the big picture of where the LLM is going.

**Comment by mik403** [comment]
Is there any update of it? I have the same thing in JetBrains Rider, and thinking about cancelling copilot subscription too, its not usable with 1 line at a time

**Comment by MildyS** [comment]
Any updated with fixing the one line suggestion to multiline suggestion in PyCharm? Or will be this update even integrated in future? Thanks.

**Comment by techdragon** [comment]
This is more than inconvenient, sometimes the code suggestions are clearly suffering. In particular, if you write a big descriptive doc-string  at the top of a brand new python module, you get it suggesting one import at a time, and it sometimes gets into a loop where it will just keep suggesting new imports, which is obviously useless behaviour.
